### PR TITLE
Fix endpoint incompatibilities

### LINKED ISSUE #91: Endpoint incompatibilities
I've made a snapshot of the JS code from MiniReg:

[minireg-dump.zip](https://github.com/DroneDB/Registry/files/5626394/minireg-dump.zip)

If you unzip the files in `Registry\Registry.Web\webroot` (replacing index.html) you can start to check for differences.

The first one I found is `GET /orgs/:org/ds/:ds`, which returns a single object:

```
{ entry }
```

Instead of an array of size 1 containing the entry:

```
[{entry}]
```

There might be other differences.

To get the routes to work properly with browser refresh, you might need to add the following equivalent (they all serve the webroot)

https://github.com/DroneDB/MiniReg/blob/master/index.js#L91-L99

### PR DESCRIPTION
**Pull request by HeDo88TH**
Closes #91

### CODE REVIEW

**Comment by pierotofy** [comment]
Have you done this?

> I've made a snapshot of the JS code from MiniReg:
> minireg-dump.zip
> If you unzip the files in Registry\Registry.Web\webroot (replacing index.html) you can start to check for differences.

**Comment by HeDo88TH** [comment]
Really annoying day:
Minireg does not start, after

```
# git clone https://github.com/DroneDB/MiniReg --recurse-submodules && cd MiniReg
# npm install
# node index.js
```

it says

```
C:\Users\IT 2\source\repos\MiniReg\node_modules\better-sqlite3\lib\database.js:32
                throw new TypeError('Cannot open database because the directory does not exist');    
                ^

TypeError: Cannot open database because the directory does not exist
    at Database (C:\Users\IT 2\source\repos\MiniReg\node_modules\better-sqlite3\lib\database.js:32:9)
    at Object.<anonymous> (C:\Users\IT 2\source\repos\MiniReg\libs\db.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1220:10)
    at Module.load (internal/modules/cjs/loader.js:1049:32)
    at Function.Module._load (internal/modules/cjs/loader.js:937:14)
    at Module.require (internal/modules/cjs/loader.js:1089:19)
    at require (internal/modules/cjs/helpers.js:73:18)
    at Object.<anonymous> (C:\Users\IT 2\source\repos\MiniReg\libs\users.js:1:16)
    at Module._compile (internal/modules/cjs/loader.js:1200:30)
```

I succeeded in starting the minireg webapp and connected to Registry. It starts doing strange things like calling:

` https://localhost:44359/orgs/orgs/ds/admin/list` 

after correctly calling

`https://localhost:44359/orgs/admin/ds/0526bvxn7ons55u2`

I don't know why. It correctly lists the organizations.

**Comment by pierotofy** [comment]
> I don't know why. It correctly lists the organizations.

Registry's `ddb://` protocol format definition is wrong.

**Comment by pierotofy** [comment]
> TypeError: Cannot open database because the directory does not exist

There needs to be a `data` directory in the root (I modified the code to auto-create it on startup). So that was a bug.

**Comment by HeDo88TH** [comment]
It is almost ready for review but I'm still figuring out why, from time to time, a logged in user receives unauthorized responses (user is null) when calling endpoints

**Comment by HeDo88TH** [comment]
Now it's ready for review 👍

File: Registry.Web/appsettings.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
     "Secret": "a2780070a24cfcaf5a4a43f931200ba0d19d8b86b3a7bd5123d9ad75b125f480fcce1f9b7f41a53abe2ba8456bd142d38c455302e0081e5139bc3fc9bf614497",
     "TokenExpirationInDays": 7,
     "RevokedTokens": [],
-    "AuthProvider": "Sqlite",
-    "RegistryProvider": "Sqlite",
+    "AuthProvider": "Mysql",
```
Comment (pierotofy) [inline Registry.Web/appsettings.json:6]: This shouldn't be the default?

**Review by pierotofy — COMMENTED** [review]
As a general rule, the program should be setup in `development` mode when you check it out from GitHub. Production mode (with mysql, redis, etc.) should be offered as a secondary option (which requires some configuration, or via docker).

Currently I can't launch this because I don't have all the database setup.

File: Registry.Web/appsettings.json (hunk 1 above)
Comment (HeDo88TH) [reply Registry.Web/appsettings.json]: I think we should leave sqlite as default. When in production we switch to mysql with caching support.

**Comment by pierotofy** [comment]
`chattr` is not working (always returns `false`).

**Comment by HeDo88TH** [comment]
There was a problem in form-data parameter deserialization, now I tested it and it's ok!

**Pull request merged**